drools-wb-screens/drools-wb-scenario-simulation-editor/drools-wb-scenario-simulation-editor-client/src/main/java/org/drools/workbench/screens/scenariosimulation/client/commands/actualcommands/DeleteColumnCommand.java

```diff
@@ -13,65 +13,46 @@
-        this.columnGroup = columnGroup;
-        this.scenarioGridPanel = scenarioGridPanel;
-        this.scenarioGridLayer = scenarioGridLayer;
-    }
+public class DeleteColumnCommand extends AbstractScenarioSimulationCommand {
 
     @Override
-    public void execute() {
-        model.deleteColumn(columnIndex);
-        if (model.getGroupSize(columnGroup) < 1) {
-            FactMappingType factMappingType = FactMappingType.valueOf(columnGroup.toUpperCase());
-            Map.Entry<String, String> validPlaceholders = model.getValidPlaceholders();
+    protected void internalExecute(ScenarioSimulationContext context) {
+        context.getModel().deleteColumn(context.getColumnIndex());
+        if (context.getModel().getGroupSize(context.getColumnGroup()) < 1) {
+            FactMappingType factMappingType = FactMappingType.valueOf(context.getColumnGroup().toUpperCase());
+            Map.Entry<String, String> validPlaceholders = context.getModel().getValidPlaceholders();
             String instanceTitle = validPlaceholders.getKey();
             String propertyTitle = validPlaceholders.getValue();
-            model.insertColumn(columnIndex, getScenarioGridColumnLocal(instanceTitle,
-                                                                       propertyTitle,
-                                                                       String.valueOf(new Date().getTime()),
-                                                                       columnGroup,
-                                                                       factMappingType,
-                                                                       scenarioGridPanel,
-                                                                       scenarioGridLayer,
-                                                                       ScenarioSimulationEditorConstants.INSTANCE.defineValidType()));
+            context.getModel().insertColumn(context.getColumnIndex(), getScenarioGridColumnLocal(instanceTitle,
+                                                                                                 propertyTitle,
+                                                                                                 String.valueOf(new Date().getTime()),
+                                                                                                 context.getColumnGroup(),
+                                                                                                 factMappingType,
+                                                                                                 context.getScenarioGridPanel(),
+                                                                                                 context.getScenarioGridLayer(),
+                                                                                                 ScenarioSimulationEditorConstants.INSTANCE.defineValidType()));
+        }
+        // TODO CHECK POSITION
```

Comment:
Do you want to add a check?

Reply:
@danielezonca 
All this " TOODO CHECK POSITION" are meant to check if the below code is in the right/best position, or should be moved elsewhere